samples/dnn/mask_rcnn.cpp

```diff
@@ -0,0 +1,204 @@
+// Deep learning based Object Detection and Instance Segmentation using Mask R-CNN
+// VIDEO 360 DEMO: https://www.youtube.com/watch?v=0tU8991QgE8
+// Downloads:
+// http://download.tensorflow.org/models/object_detection/mask_rcnn_inception_v2_coco_2018_01_28.tar.gz
+// https://raw.githubusercontent.com/opencv/opencv_extra/master/testdata/dnn/mask_rcnn_inception_v2_coco_2018_01_28.pbtxt
+// https://raw.githubusercontent.com/spmallick/learnopencv/master/Mask-RCNN/mscoco_labels.names
+// It is based on the OpenCV project. It is subject to the license terms in the LICENSE file found in this distribution and at http://opencv.org/license.html
+//
+
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <string>
+
+#include <opencv2/dnn.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+
+using namespace cv;
+using namespace dnn;
+using namespace std;
+
+string keys =
+    "{ help  h     | | Print help message. \nUsage \n\t\t./mask_rcnn --image=logo.jpg \n\t\t ./mask_rcnn --video=teste.mp4}"
+    "{ image m     |<none>| Path to input image file.  }"
+    "{ video v     |<none>| Path to input video file.  }"
+    "{ cam  C      |0| Path to input video file.  }"
+    "{ classes n   |mscoco_labels.names | Path to a text file with names of classes.  }"
```

Comment:
I think it's better to use default name `object_detection_classes_coco.txt`. This is a name of file which might be found in OpneCV's source code tree: https://github.com/opencv/opencv/blob/master/samples/data/dnn/object_detection_classes_coco.txt.

Reply:
done... thank you